src/components/WriteEditModal/WriteEditModal.tsx

```diff
@@ -1,39 +1,164 @@
-import KEYBOARD_DATA from '@/app/(test)/mj/customData';
-import { Button, ImageInput, InputField, KeyboardInfoBox, TextField } from '@/components';
+'use client';
+
 import classNames from 'classnames/bind';
-import { ChangeEvent, useState } from 'react';
+import Image from 'next/image';
+import { useForm, SubmitHandler, FieldValues } from 'react-hook-form';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { CommunityPostCardDetailDataType, PostCardDetailModalCustomKeyboardType } from '@/types/CommunityTypes';
+import { Button, ImageInput, InputField, TextField } from '@/components';
+import { keydeukImg } from '@/public/index';
+import { postCreateCustomReview, putEditCustomReview } from '@/api/communityAPI';
+
+import { toast } from 'react-toastify';
+import { useState } from 'react';
 import styles from './WriteEditModal.module.scss';
 
 const cn = classNames.bind(styles);
 
 interface WriteEditModalProps {
+  keyboardInfo: PostCardDetailModalCustomKeyboardType;
   isCustomReview?: boolean;
+  editCustomData?: CommunityPostCardDetailDataType;
+  onSuccessReview: () => void;
 }
 
 const TITLE_MAX_LENGTH = 20;
 const TITLE_PLACEHOLDER = '미 입력 시 키득 커스텀 키보드로 등록됩니다.';
 const CONTENT_PLACEHOLDER = '최소 20자 이상 입력해주세요';
 
-export default function WriteEditModal({ isCustomReview }: WriteEditModalProps) {
-  const [title, setTitle] = useState('');
-  const [content, setContent] = useState('');
+export default function WriteEditModal({
+  keyboardInfo,
+  isCustomReview,
+  editCustomData,
+  onSuccessReview,
+}: WriteEditModalProps) {
+  const queryClient = useQueryClient();
 
-  const handleTitleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setTitle(e.target.value);
+  const [deletedImageId, setDeleteImageId] = useState<number[]>([]);
+
+  const handleSaveDeletedImageId = (id: number) => {
+    setDeleteImageId((prevIds) => [...prevIds, id]);
   };
 
-  const handleContentChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
-    setContent(e.target.value);
+  const defaultImages = [];
+
+  if (editCustomData) {
+    for (let i = 0; i < editCustomData.reviewImages.length; i += 1) {
+      defaultImages.push(editCustomData.reviewImages[i].imgUrl);
+    }
```

Comment:
forEach사용하면 더 간결해지지 않을까요?
```suggestion
    editCustomData.reviewImages.forEach((image) => (defaultImage.push(image.imageUrl)))
```

push를 한 번에 처리하는 것도?

```suggestion
    const ImageUrlList = editCustomData.reviewImages.map((image) => image.Url);
    defaultImage.push(...ImageUrlList));
```